collector/zfs.go

```diff
@@ -65,6 +65,14 @@ func NewZFSCollector(logger log.Logger) (Collector, error) {
 }
 
 func (c *zfsCollector) Update(ch chan<- prometheus.Metric) error {
+
+	if _, err := c.openProcFile(c.linuxProcpathBase); err != nil {
```

Comment:
I think it would make more sense to test this in NewZFSCollector, no?

Reply:
Hmm... Moving the check to NewZFSCollector and returning an error results in panic and program termination when starting the node_exporter (since zfs is default enabled). The erros are as follows: 
```
panic: Couldn't create metrics handler: couldn't create collector: failed to open zfs: open /proc/spl/kstat/zfs: no such file or directory

goroutine 1 [running]:
main.newHandler(...)
        node_exporter.go:63 
main.main()
        node_exporter.go:180 
```

Reply:
Ah I see.. yeah it looks like we fail everything if thew New method of a collector returns an error: https://github.com/prometheus/node_exporter/blob/f07e982c77ddb3b042ccfa895139c5c743962a9a/collector/collector.go#L122

Now I'm thinking.. maybe we should handle ErrNoData there (log it, move on). @SuperQ wdyt?